enclave-runtime/src/lib.rs

```diff
@@ -726,51 +729,52 @@ where
 	Signer::Signature: Into<MultiSignature>,
 {
 	let mut calls = Vec::<OpaqueCall>::new();
-	let mrenclave: ShardIdentifier = on_chain_ocall_api.get_mrenclave_of_self()?.m.into();
 
 	debug!("Syncing light client!");
 	for signed_block in blocks_to_sync.into_iter() {
-		validator
-			.check_xt_inclusion(validator.num_relays(), &signed_block.block)
-			.unwrap(); // panic can only happen if relay_id does not exist
+		let block = signed_block.block;
+		validator.check_xt_inclusion(validator.num_relays(), &block).unwrap(); // panic can only happen if relay_id does not exist
 
 		if let Err(e) = validator.submit_simple_header(
 			validator.num_relays(),
-			signed_block.block.header().clone(),
+			block.header().clone(),
 			signed_block.justifications.clone(),
 		) {
 			error!("Block verification failed. Error : {:?}", e);
 			return Err(e.into())
 		}
 
-		if let Err(e) = stf_executor.update_states::<PB>(&signed_block.block.header()) {
+		if let Err(e) = stf_executor.update_states::<PB>(&block.header()) {
 			error!("Error performing state updates upon block import");
 			return Err(e.into())
 		}
 
 		// execute indirect calls, incl. shielding and unshielding
-		match scan_block_for_relevant_xt(&signed_block.block, stf_executor) {
-			// push shield funds to opaque calls
-			Ok(c) => calls.extend(c.into_iter()),
+		match scan_block_for_relevant_xt(&block, stf_executor) {
+			Ok((unshielding_call_confirmations, executed_shielding_calls)) => {
+				// Include all unshieldung confirmations that need to be executed on the parentchain.
+				calls.extend(unshielding_call_confirmations.into_iter());
+				// Include a processed parentchain block confirmation for each block.
+				calls.push(processed_parentchain_block_extrinsic(
+					block.hash(),
+					executed_shielding_calls,
+				));
+			},
 			Err(_) => error!("Error executing relevant extrinsics"),
 		};
-
-		// compose indirect block confirmation
-		// should be changed to ParentchainBlockProcessed, see worker issue #457
-		let xt_block = [TEEREX_MODULE, BLOCK_CONFIRMED];
-		let block_hash = signed_block.block.header().hash();
-		let prev_state_hash = signed_block.block.header().parent_hash();
-		calls.push(OpaqueCall::from_tuple(&(
-			xt_block,
-			mrenclave, // 'block_confirmed' only accepts shard == mrenclave. Overall 'block_confirmed' construct will be adjusted with #457
-			block_hash,
-			prev_state_hash.encode(),
-		)));
 	}
 
 	prepare_and_send_xts::<_, _, _, Signer>(validator, signer, on_chain_ocall_api, calls, nonce)
 }
 
+/// Creates a processed_parentchain_block extrinsic for a given parentchain block hash and the merkle executed extrinsics.
+///
+/// Calculates the merkle root of the extrinsics. In case no extrinsics are supplied, the root will be a hash filled with zeros.
+fn processed_parentchain_block_extrinsic(block_hash: H256, extrinsics: Vec<H256>) -> OpaqueCall {
```

Comment:
Maybe name this function `create_processed_parentchain_block_extrinsic` to be clear what it does?

Reply:
Done 👍

Reply:
Renamed it to `create_processed_parentchain_block_call` to be clear, extrinsics will only be created later on.